include/mxnet/ndarray.h

```diff
@@ -507,6 +507,35 @@ class NDArray {
     ret.reuse_ = true;
     return ret;
   }
+
+  inline void SparseUpdateChunk(const NDArray &arr) const {
+    auto stype = arr.storage_type();
+    CHECK(stype == kCSRStorage || stype == kRowSparseStorage)
+        << "Only to be used with CSR and RSP storage types";
+    ptr_->shandle.dptr = arr.ptr_->shandle.dptr;
```

Comment:
Would `ptr_->shandle = arr.ptr_->shandle ` be sufficient?

Reply:
this doesn't work because the Handle struct also stores a pointer to data. doing `ptr_->shandle = arr.ptr_->shandle` would make copies of dptr which point to same data. But then sparse updates the dptr at runtime and this wont reflect in the copied shandle.